### PR TITLE
fix: migrate buttons 0.12.0

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a link to the associated issue.
part of https://github.com/SAP/fundamental-ngx/issues/3370

#### Please provide a brief summary of this pull request.
There is new way to add text inside button component. The `[label]` property should be used now instead of adding text as a content. The content still works fine, unless it's used with icons.

BREAKING CHANGE:

The value for the text button is now passed as an input property, not as content projection.

Before: `<button fd-button>Text</button>`
After: `<button fd-button [label]="'Text'"></button`
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by JKMarkowski** [comment]
image

File: libs/core/src/lib/button/button.component.ts [hunk 1]
```diff
@@ -74,12 +82,22 @@ export class ButtonComponent implements OnChanges, CssClassBuilder, OnInit {
...
+
     /** Whether to apply menu mode to the button.
      * Default value is set to false
      */
     @Input()
     public fdMenu = false;
 
+    /** @hidden */
+    @ViewChild('textElementRef')
+    textElementRef: ElementRef;
```
Comment (salarenko) [inline libs/core/src/lib/button/button.component.ts:99]: Is this needed?

File: libs/core/src/lib/button/button.component.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+<ng-container *ngIf="glyphPosition === 'after' && glyph">
+    <ng-container *ngTemplateOutlet="iconTemplate"></ng-container>
+</ng-container>
+
+<fd-icon glyph="slim-arrow-down" *ngIf="fdMenu" role="presentation"></fd-icon>
+
+<ng-template #iconTemplate>
+    <fd-icon
+        role="presentation"
```
Comment (salarenko) [inline libs/core/src/lib/button/button.component.html:19]: `fd-icon` by default provides `role="presentation"`

**Comment by JKMarkowski** [comment]
Time increase/decrease button doesn't have icon
image

File: libs/core/src/lib/button/button.component.ts (hunk 1 above)
Comment (JKMarkowski) [reply libs/core/src/lib/button/button.component.ts:99]: It should be removed

**Comment by mikerodonnell89** [comment]
Dialogs do have buttons now but they are not centered

Screen Shot 2020-09-28 at 1 46 23 PM

**Comment by JKMarkowski** [comment]
@mikerodonnell89 Dialog close button has been fixed

**Review by mikerodonnell89 — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

File: apps/docs/src/app/core/component-docs/menu/examples/menu-separator-example.component.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <div style="display: inline-block; width: 50%;">
-    <button fd-button [fdMenu]="true" [fdMenuTrigger]="menu">Menu</button>
+    <button fd-button label="Menu" [fdMenu]="true" [fdMenuTrigger]=""></button>
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/menu/examples/menu-separator-example.component.html:2]: You broke menu example here `[fdMenuTrigger]=""`

**Review by salarenko — CHANGES_REQUESTED** [review]

**Review by salarenko — APPROVED** [review]

**Pull request merged**